addons/crm/models/crm_lead.py

```diff
@@ -714,7 +714,9 @@ def create(self, vals_list):
             if vals.get('website'):
                 vals['website'] = self.env['res.partner']._clean_website(vals['website'])
         leads = super(Lead, self).create(vals_list)
-
+        odoobot = self.env.ref('base.user_root')
+        assigned_team_leads = leads.filtered(lambda lead: lead.team_id and lead.user_id != odoobot)
+        assigned_team_leads._compute_team_id()  # Ensure the sales team matches the salesperson assigned to the lead
```

Comment:
Seems sub-optimal to always re-assign the team, when maybe I just wanted to pick a team manually.